pkg/kptfile/kptfileutil/util.go

```diff
@@ -224,6 +225,11 @@ func UpdateUpstreamLockFromGit(path string, spec *git.RepoSpec) error {
 }
 
 func merge(localKf, updatedKf, originalKf *kptfilev1.KptFile) error {
```

Comment:
we should add pointer to the design doc here, will be very useful for future contributors :)

Reply:
added